code/game/machinery/vitals_monitor.dm

```diff
@@ -50,76 +72,157 @@
+	else
+		visible_message(SPAN_NOTICE("\The [src] is no longer relaying data from a connected operating table."))
+
 /obj/machinery/vitals_monitor/MouseDrop(over_object, src_location, over_location)
 	if(!CanMouseDrop(over_object))
 		return
+	update_optable()
 	if(victim)
-		victim = null
-		update_use_power(POWER_USE_IDLE)
+		update_victim()
 	else if(ishuman(over_object))
-		victim = over_object
-		update_use_power(POWER_USE_ACTIVE)
-		visible_message(SPAN_NOTICE("\The [src] is now showing data for [victim]."))
+		update_victim(over_object)
+	else if(istype(over_object, /obj/machinery/optable/))
+		var/obj/machinery/optable/new_table_connection = over_object
+		update_optable(new_table_connection)
 
 /obj/machinery/vitals_monitor/on_update_icon()
 	overlays.Cut()
 	if(stat & NOPOWER)
 		return
 	overlays += image(icon, icon_state = "screen")
 
+	handle_pulse()
+	handle_brain()
+	handle_lungs()
+	handle_alerts()
+
+/obj/machinery/vitals_monitor/proc/handle_pulse()
 	if(!victim)
 		return
-	
-	switch(victim.pulse())
-		if(PULSE_NONE)
-			overlays += image(icon, icon_state = "pulse_flatline")
-			overlays += image(icon, icon_state = "pulse_warning")
-		if(PULSE_SLOW, PULSE_NORM,)
-			overlays += image(icon, icon_state = "pulse_normal")
-		if(PULSE_FAST, PULSE_2FAST)
-			overlays += image(icon, icon_state = "pulse_veryfast")
-		if(PULSE_THREADY)
-			overlays += image(icon, icon_state = "pulse_thready")
-			overlays += image(icon, icon_state = "pulse_warning")
+	var/obj/item/organ/internal/heart/heart = victim.internal_organs_by_name[BP_HEART]
+	if(istype(heart) && !BP_IS_PROSTHETIC(heart))
+		switch(victim.pulse())
+			if(PULSE_NONE)
+				overlays += image(icon, icon_state = "pulse_flatline")
+				overlays += image(icon, icon_state = "pulse_warning")
+				if(beep)
+					playsound(src, 'sound/machines/flatline.ogg', 20)
+				if(read_alerts)
+					alerts[PULSE_ALERT] = "Cardiac flatline detected!"
+			if(PULSE_SLOW, PULSE_NORM,)
+				overlays += image(icon, icon_state = "pulse_normal")
+				if(beep)
+					playsound(src, 'sound/machines/quiet_beep.ogg', 40)
+			if(PULSE_FAST, PULSE_2FAST)
+				overlays += image(icon, icon_state = "pulse_veryfast")
+				if(beep)
+					playsound(src, 'sound/machines/quiet_double_beep.ogg', 40)
+			if(PULSE_THREADY)
+				overlays += image(icon, icon_state = "pulse_thready")
+				overlays += image(icon, icon_state = "pulse_warning")
+				if(beep)
+					playsound(src, 'sound/machines/ekg_alert.ogg', 40)
+				if(read_alerts)
+					alerts[PULSE_ALERT] = "Excessive heartbeat! Possible Shock Detected!"
+	else
+		overlays += image(icon, icon_state = "pulse_warning")
 
+/obj/machinery/vitals_monitor/proc/handle_brain()
+	if(!victim)
+		return
 	var/obj/item/organ/internal/brain/brain = victim.internal_organs_by_name[BP_BRAIN]
 	if(istype(brain) && victim.stat != DEAD && !(victim.status_flags & FAKEDEATH))
 		switch(brain.get_current_damage_threshold())
 			if(0 to 2)
 				overlays += image(icon, icon_state = "brain_ok")
```

Comment:
Not sure why these are procs, they are only used to update the icon and will break overlays if called outside of it.